component/prometheus/exporter/mysql/mysql.go

```diff
@@ -23,6 +27,26 @@ func createExporter(opts component.Options, args component.Arguments) (integrati
 	return a.Convert().NewIntegration(opts.Logger)
 }
 
+func customizeTarget(baseTarget discovery.Target, args component.Arguments) []discovery.Target {
+	a := args.(Arguments)
+	target := baseTarget
+
+	m, err := mysql.ParseDSN(string(a.DataSourceName))
```

Comment:
It feels a bit odd to do this level of validation here, that the component will also do on update. I don't have a super good alternative though, if the arguments are all we are given. Will the unmarshalRiver validation handle this as well?

Reply:
yeah it is weird and kind of ugly, currently `ParseDSN` is only called in `go-sql-driver`, I think it is a good idea to also call it early as a validation